src/acl/Url.cc

```diff
@@ -11,16 +11,15 @@
 #include "squid.h"
 #include "acl/FilledChecklist.h"
 #include "acl/Url.h"
+#include "anyp/Uri.h"
```

Comment:
```diff
- Maintenance: adopt AnyP::Uri::Decode [1/n]
+ Use AnyP::Uri::Decode() for url_regex checks
```

I have adjusted PR title to remove the implication that these changes have no important runtime effects: The new Decode() is quite different from the old rfc1738_unescape() in its treatment of invalid encodings! I also removed `[1/n]` counter because it has virtually no useful purpose[^1] while requiring maintenance/synchronization across PRs, potentially creating conflicts and disagreements.

This comment does not request any PR changes but please keep it in mind when working on other PRs.

[^1]: One can find relevant PRs/commits by searching for "Use AnyP::Uri::Decode" and "Drop rfc1738_unescape" in titles and ordering those PRs/commits any way they want.